src/Controls/src/Core/DragAndDrop/PlatformDropEventArgs.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+
+namespace Microsoft.Maui.Controls;
+
+/// <summary>
+/// Platform-specific arguments associated with the DropEventArgs.
+/// </summary>
+public class PlatformDropEventArgs
+{
+#if IOS || MACCATALYST
+	/// <summary>
+	/// Gets the native view attached to the event.
+	/// </summary>
+	public UIKit.UIView? Sender { get; }
+
+	/// <summary>
+	/// Gets the interaction used for dropping items.
+	/// </summary>
+	public UIKit.UIDropInteraction DropInteraction { get; }
+
+	/// <summary>
+	/// Gets the associated information from the drop session.
+	/// </summary>
+	public UIKit.IUIDropSession DropSession { get; }
+
+	internal PlatformDropEventArgs(UIKit.UIView? sender, UIKit.UIDropInteraction dropInteraction,
+		UIKit.IUIDropSession dropSession)
+	{
+		Sender = sender;
+		DropInteraction = dropInteraction;
+		DropSession = dropSession;
+	}
+
+#elif ANDROID
+	/// <summary>
+	/// Gets the native view attached to the event.
+	/// </summary>
+	public Android.Views.View Sender { get; }
+
+	/// <summary>
+	/// Gets the event containing information for drag and drop status.
+	/// </summary>
+	public Android.Views.DragEvent DragEvent { get; }
+
+	internal PlatformDropEventArgs(Android.Views.View sender, Android.Views.DragEvent dragEvent)
+	{
+		Sender = sender;
+		DragEvent = dragEvent;
+	}
+
+#elif WINDOWS
+	/// <summary>
+	/// Gets the native view attached to the event.
+	/// </summary>
+	public object Sender { get; }
```

Comment:
Should this be `Microsoft.UI.Xaml.UIElement`?

Reply:
Some of the calling methods used just have their sender type as object as such:
```cs
void HandleDrop(object sender, Microsoft.UI.Xaml.DragEventArgs e)
{ ... }
```

Reply:
I could type cast and check for Microsoft.UI.Xaml.UIElement, but I was not sure if we expect that every time or if there was a reason for the 'object'

Reply:
Sender should be the object you are attaching to, so if they are all UIElement, maybe we can keep it consistent

Reply:
Same as above, this probably should be UIElement - even though it is object. The events are only available on UIElement so should never be anything other than element.